[RFT] [24.10] mac80211: update to v6.14.11

namiltd opened the pull request:
Just for testing!

Update Backports mac80211 to version 6.14.11 for openwrt-24.10 branch.
All commits that determine the transition to the new backports version in the main branch have been added.

Tested successfully on routers:

 -   ramips/mt7620 - TP-Link Archer C5 v4 (kernel 6.6.93):
```
[   21.363393] Loading modules backported from Linux version v6.14.11-0-gb9d5d463c216
[   21.371157] Backport generated by backports.git v6.1.110-1-43-g2e32643b84e8
[   22.292987] mt76x2e 0000:01:00.0: card - bus=0x1, slot = 0x0 irq=4
[   22.299639] mt76x2e 0000:01:00.0: ASIC revision: 76120044
[   23.003427] mt76x2e 0000:01:00.0: ROM patch build: 20141115060606a
[   23.034327] mt76x2e 0000:01:00.0: Firmware Version: 0.0.00
[   23.039969] mt76x2e 0000:01:00.0: Build: 1
[   23.044227] mt76x2e 0000:01:00.0: Build Time: 201607111443____
[   23.112251] mt76x2e 0000:01:00.0: Firmware running!
[   23.117433] mt76x2e 0000:01:00.0: registering led 'mt76-phy0'
```

georgemoussalem committed:
mac80211: ath11k: Support setting bdf-addr and caldb-addr via DT

IPQ5018 uses different BDF and caldb addresses for vairous boards,
so let's support reading these addresses from the device tree.

Signed-off-by: Ziyang Huang <hzyitc@outlook.com>
Signed-off-by: George Moussalem <george.moussalem@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/17182
Signed-off-by: Robert Marko <robimarko@gmail.com>
(cherry picked from commit 432f2f83decb826df7e20324919a32bd4491f862)

georgemoussalem committed:
mac80211: ath11k: add support for QCN6122 wifi

Add QCN6122 platform support.

QCN6122 is a PCIe based solution that is attached to and enumerated
by the WPSS (Wireless Processor SubSystem) Q6 processor.

Though it is a PCIe device, since it is not attached to APSS processor
(Application Processor SubSystem), APSS will be unaware of such a decice
and hence it is registered to the APSS processor as a platform device(AHB).
Because of this hybrid nature, it is called as a hybrid bus device.

As such, QCN6122 is a hybrid bus type device and follows the same codepath
as for WCN6750.

This is a reversed engineered and heavily simplified version of below
downstream patch:
https://git.codelinaro.org/clo/qsdk/oss/system/feeds/wlan-open/-/ \
blob/NHSS.QSDK.12.4.5.r2/mac80211/patches/232-ath11k-qcn6122-support.patch

Co-developed-by: George Moussalem <george.moussalem@outlook.com>
Signed-off-by: Sowmiya Sree Elavalagan <ssreeela@codeaurora.org>
Signed-off-by: George Moussalem <george.moussalem@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/17182
Signed-off-by: Robert Marko <robimarko@gmail.com>
(cherry picked from commit f76b846eea288be452be070413353185e7abe822)

georgemoussalem committed:
mac80211: ath11k: fix remapped ce access on 64-bit OS

https://lore.kernel.org/linux-wireless/TYZPR01MB55563B3A689D54D18179E5B4C9192@TYZPR01MB5556.apcprd01.prod.exchangelabs.com/

Signed-off-by: Ziyang Huang <hzyitc@outlook.com>
Signed-off-by: George Moussalem <george.moussalem@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/17182
Signed-off-by: Robert Marko <robimarko@gmail.com>
(cherry picked from commit f520f54ab261677863f2e071da9ba5b1960227ad)

georgemoussalem committed:
mac80211: ath11k: poll reo status ring for IPQ5018

This downstream patch fixes a bug which could flood the logs with the
following message and would eventually lead to a crash.

ath11k c000000.wifi: failed to send HAL_REO_CMD_UPDATE_RX_QUEUE cmd, tid 0 (-105)

Signed-off-by: George Moussalem <george.moussalem@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/17182
Signed-off-by: Robert Marko <robimarko@gmail.com>
(cherry picked from commit c9c0f1d8e5d1f97d6c2f2251b101606c3493fcfb)

neheb committed:
mac80211: ath9k: clean up gpiochip

Simplify patch with gpiochip_add_data, struct reduction, new GPIO API,
and header cleanup.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/17445
Signed-off-by: Robert Marko <robimarko@gmail.com>
(cherry picked from commit e78dc2eae49c5980a5201c98b26cd4a1b177c9a5)

neheb committed:
mac80211: ath9k: remove platform leds

These only work with and are useful with mach files. Now that those are
gone, this can go too.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/17445
Signed-off-by: Robert Marko <robimarko@gmail.com>
(cherry picked from commit baf73daaacdef46307454bc5d507c31d9033c01c)

neheb committed:
mac80211: ath9k: remove gpio buttons support

This is only used by mach files, which are no longer used in OpenWrt.

Allows removing a custon ath9k_platform.h file.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/17445
Signed-off-by: Robert Marko <robimarko@gmail.com>
(cherry picked from commit d70f8dea1ec5d5de34cf294d42520747247de387)

guidosarducci committed:
mac80211: Fix tracing backport

Update handling of macro __assign_str() to also support the one-argument
variant when building against kernel 6.10 or later. This is needed for
building the next LTS kernel 6.12.

Fixes: 384d079fd8 ("mac80211: update to version 6.11")
Signed-off-by: Tony Ambardar <itugrok@yahoo.com>
Link: https://github.com/openwrt/openwrt/pull/17456
Signed-off-by: Robert Marko <robimarko@gmail.com>
(cherry picked from commit 9096f996baf5555bae1c3175a9ddf7bbd3be9a9c)

neheb committed:
mac80211: fix led-pin usage

ath9k uses a deprecated GPIO API that assumes a starting base of 0. This
is no longer the case and so must be fixed.

Upstream pending patch.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/17445
Signed-off-by: Robert Marko <robimarko@gmail.com>
(cherry picked from commit 8534844847d47d1f883012c0297422d5611c7e83)

neheb committed:
mac80211: ath9k: simplify probe with devm

Upstream backport.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/17445
Signed-off-by: Robert Marko <robimarko@gmail.com>
(cherry picked from commit 410111a8daadd9e71fddc4ea278ce0a372b863ad)

nbd168 committed:
mac80211: fix issues with including linux/acpi_amd_wbrf.h on newer kernels

Change the double-include guard to avoid conflict with the linux kernel header

Reported-by: Tony Ambardar <itugrok@yahoo.com>
Signed-off-by: Felix Fietkau <nbd@nbd.name>
(cherry picked from commit 7bf3bc871ac1ae6b6f199ba1bd83eba9d017a0d5)

nbd168 committed:
mac80211: backport API change for sta rate control update

Helps stay in sync with mt76 upstream development

Signed-off-by: Felix Fietkau <nbd@nbd.name>
(cherry picked from commit bfb106e8ae7390ad920ef1c40e78da22a88b2c7d)

mantas-p committed:
mac80211: ath12k: support fetching regdb from board-2.bin

In board-2.bin available at linux-firmare regdb is stored with
board-id=255. This change is needed to properly use it.

Signed-off-by: Mantas Pucka <mantas@8devices.com>
Link: https://github.com/openwrt/openwrt/pull/18459
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
(cherry picked from commit c1acef2e0cb2fcaf1ee72dbf737f6017ddb1159e)

DragonBluep committed:
ath10k-ct: update to version 6.14

This new version fixes some memory leak and NULL pointer issues.

Upstreamed patches:
[1] 010-api_update.patch
[2] 201-wifi-ath10k-add-LED-and-GPIO-controlling-support-for.patch

The new 003 patch was introduced to fix the build error on old LTS
kernel. The patch 004 was used to fix the issue of IPQ4019 radio
not being able to start.

Tested on QCA9882 and IPQ4019.

[1] https://github.com/torvalds/linux/commit/bca8bc0399ac2efd56e6adbed0307e10125a556c
[2] https://github.com/torvalds/linux/commit/8e1debd82466a3fe711784ab37e6b54e56011267

Signed-off-by: Shiji Yang <yangshiji66@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/18368
Signed-off-by: Robert Marko <robimarko@gmail.com>
(cherry picked from commit b47cf6e2db8ea5535350f2b7368ebae66fc749c9)

mantas-p committed:
mac80211: ath12k: fix wideband operation on QCN9274

Add patches to enable QCN9274 radios that support both 5GHz and 6GHz
bands.

Signed-off-by: Mantas Pucka <mantas@8devices.com>
Link: https://github.com/openwrt/openwrt/pull/18459
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
(cherry picked from commit 49e6e1daf4ae33536b9e26c34d248ffc23c9a0b4)

mantas-p committed:
mac80211: ath12k: add patches for 160MHz support

Add series enabling 160MHz channels on ath12k

Signed-off-by: Mantas Pucka <mantas@8devices.com>
Link: https://github.com/openwrt/openwrt/pull/18459
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
(cherry picked from commit 9e3a2466f6944c77276aa6e7cd3f98dc6ffdadf1)

Ansuel committed:
ath10k-ct: partially revert removal of 003 patch

Partially revert removal of 003 patch to fix compilation error with
kernel 6.6.

In 6.6 .remove still require int and .remove_new needs to be used.

Fixes: bfb106e8ae73 ("mac80211: backport API change for sta rate control update")
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
(cherry picked from commit b68173500a1a6455a0faa3c4d54b2c62cf8593f1)

mantas-p committed:
mac80211: ath12k: fix regdb parsing for 6GHz

Add backport patch that fixes ath12k regdomain parsing failure in
6GHz band triggered by the latest regdb/board-2.bin update.

Signed-off-by: Mantas Pucka <mantas@8devices.com>
Link: https://github.com/openwrt/openwrt/pull/18512
Signed-off-by: Robert Marko <robimarko@gmail.com>
(cherry picked from commit 8ffb3b2829773455641d406eada7c9bfaf65b89e)

mantas-p committed:
mac80211: refresh patches

Previous commits missed some refreshing

Signed-off-by: Mantas Pucka <mantas@8devices.com>
Link: https://github.com/openwrt/openwrt/pull/18512
Signed-off-by: Robert Marko <robimarko@gmail.com>
(cherry picked from commit 903566c2d62a9aa98a965a9cf03fbc84d96860ba)

neheb committed:
mac80211: ath9k: remove wrong devm for request_irq

Since the irq gets copied to sc, it's a really bad idea to use devm,
especially when probe fails.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
Fixes: 4e18d22 ("mac80211: ath9k: simplify probe with devm")
Link: https://github.com/openwrt/openwrt/pull/18570
Signed-off-by: Robert Marko <robimarko@gmail.com>
(cherry picked from commit 410111a8daadd9e71fddc4ea278ce0a372b863ad)

DragonBluep committed:
mac80211: rt2x00: fix build warnings on 6.12 kernel

Move rt2x00lib_read_eeprom() function prototype from rt2800lib.h to
rt2x00.h and make rt6352_enable_pa_pin() static to fix the missing
prototypes warnings:

/workspaces/openwrt/build_dir/target-x86_64_musl/linux-x86_64/mac80211-regular/backports-6.12.6/drivers/net/wireless/ralink/rt2x00/rt2x00eeprom.c:213:5: error: no previous prototype for 'rt2x00lib_read_eeprom' [-Werror=missing-prototypes]
  213 | int rt2x00lib_read_eeprom(struct rt2x00_dev *rt2x00dev)
      |     ^~~~~~~~~~~~~~~~~~~~~
/workspaces/openwrt/build_dir/target-x86_64_musl/linux-x86_64/mac80211-regular/backports-6.12.6/drivers/net/wireless/ralink/rt2x00/rt2800lib.c:308:6: error: no previous prototype for 'rt6352_enable_pa_pin' [-Werror=missing-prototypes]
  308 | void rt6352_enable_pa_pin(struct rt2x00_dev *rt2x00dev, int enable)
      |      ^~~~~~~~~~~~~~~~~~~~

Signed-off-by: Shiji Yang <yangshiji66@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/18637
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>
(cherry picked from commit bad441ae4e60b9203eb4f6370026cc52e9b29384)

DragonBluep committed:
mac80211: fix unaligned.h header location for 6.12 kernel

Fix mt76 build error on 6.12 kernel:

In file included from /workspaces/openwrt/build_dir/target-x86_64_musl/linux-x86_64/mt76-2025.04.11~be28ef77/mt76x0/eeprom.c:13:
/workspaces/openwrt/staging_dir/target-x86_64_musl/usr/include/mac80211-backport/asm/unaligned.h:3:15: fatal error: asm/unaligned.h: No such file or directory
    3 | #include_next <asm/unaligned.h>
      |               ^~~~~~~~~~~~~~~~~

Signed-off-by: Shiji Yang <yangshiji66@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/18637
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>
(cherry picked from commit 9aea4c9280931d03464b7543cae8d727f4b1e037)

DragonBluep committed:
mac80211: ath9k: fix missing prototypes warnings

Fix following warnings by including the corresponding headers:

backports-6.12.6/drivers/net/wireless/ath/ath9k/hsr.c:50:6: error: no previous prototype for 'ath9k_hsr_init' [-Werror=missing-prototypes]
   50 | void ath9k_hsr_init(struct ath_hw *ah)
      |      ^~~~~~~~~~~~~~
backports-6.12.6/drivers/net/wireless/ath/ath9k/hsr.c:165:5: error: no previous prototype for 'ath9k_hsr_disable' [-Werror=missing-prototypes]
  165 | int ath9k_hsr_disable(struct ath_hw *ah)
      |     ^~~~~~~~~~~~~~~~~
backports-6.12.6/drivers/net/wireless/ath/ath9k/hsr.c:177:5: error: no previous prototype for 'ath9k_hsr_enable' [-Werror=missing-prototypes]
  177 | int ath9k_hsr_enable(struct ath_hw *ah, int bw, int fq)
      |     ^~~~~~~~~~~~~~~~
backports-6.12.6/drivers/net/wireless/ath/ath9k/hsr.c:234:5: error: no previous prototype for 'ath9k_hsr_status' [-Werror=missing-prototypes]
  234 | int ath9k_hsr_status(struct ath_hw *ah)
      |     ^~~~~~~~~~~~~~~~

Signed-off-by: Shiji Yang <yangshiji66@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/18637
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>
(cherry picked from commit 6deb2a03ba71c4e8d5315fbe6cbcf13874ee2f58)

namiltd committed:
mac80211: rt2x00: use remove_new

Easy compability fix for kernel 6.12.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
Link: https://github.com/openwrt/openwrt/pull/18719
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>
(cherry picked from commit 6b0f414aff8ef8bd453c3b907781979c64c3e580)

neheb committed:
mac80211: move OF stuff to ath9k_of_init

It's the proper function to handle this stuff in.

The original patch abused the fact that the ath9k driver in init called
ath9k_init_platform to populate all the needed configuration. This is
the wrong place to do so and it also goes away in 6.13.

Move 553-ath9k_of_gpio_mask.patch contents to ath9k_of_init where they
belong.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/18764
Signed-off-by: Robert Marko <robimarko@gmail.com>
(cherry picked from commit 12913c3c5658992985e13f4395dee86e5450154d)

neheb committed:
mac80211: ath9k: replace qca,led-sources

Upstream seems to be using led-sources instead of custom properties.

Code mostly taken from mt76.

Changed all(few) users of qca,led_pin to use the new format.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/18805
Signed-off-by: Robert Marko <robimarko@gmail.com>
(cherry picked from commit 3d6e1a3c9d2e0fb4b2e0dde7fabd269c7769db2d)

nbd168 committed:
mac80211: allow scanning operating on DFS channels

Only for multi-radio configurations where the scan would not affect the radio
running DFS detection

Signed-off-by: Felix Fietkau <nbd@nbd.name>
(cherry picked from commit 91d33d156349796c88f9e4fafd4de2581b2a2d8b)

namiltd committed:
mac80211: update to version 6.14.11

Update to version 6.14.11

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
Signed-off-by: Felix Fietkau <nbd@nbd.name>
(cherry picked from commit 88f4c32060f79e5161a789e85da299cebbde68bf)

Noltari committed:
generic: 6.6: backport devm_clk_get_optional_enabled_with_rate

This is needed for mac80211 v6.14.5 update.

9934a1bd45b2b clk: provide devm_clk_get_optional_enabled_with_rate()

https://github.com/torvalds/linux/commit/9934a1bd45b2b03f6d1204a6ae2780d3b009799f

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>
(cherry picked from commit 268c352e44956a3e7fe86e68ad727ba57b0a75a5)

Noltari committed:
generic: backport secs_to_jiffies

This is needed for mac80211 v6.14.5 update.

b35108a51cf7b jiffies: Define secs_to_jiffies()
bb2784d9ab495 jiffies: Cast to unsigned long in secs_to_jiffies() conversion

https://github.com/torvalds/linux/commit/b35108a51cf7bab58d7eace1267d7965978bcdb8
https://github.com/torvalds/linux/commit/bb2784d9ab49587ba4fbff37a319fff2924db289

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>
(cherry picked from commit f40250781de03b7c6e789645016e19def93563d0)

neheb committed:
mac80211: ath9k: fix refclock typo

Fixes ath9k on ar934x platforms.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/19031
Signed-off-by: Robert Marko <robimarko@gmail.com>
(cherry picked from commit 6f607c69dc39c3532084a46be80287932f8bd4d2)

neheb committed:
mac80211: ath9k: avoid a double reset

The original OF code effectively does a reset at ahb.c but then again in
hw.c. For AR9330, it's already done in the driver and with the others,
there are patches in here that do the same. hw.c looks like the proper
place to handle this.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/19031
Signed-off-by: Robert Marko <robimarko@gmail.com>
(cherry picked from commit 0ccc336c56fca7ecaede590995da30889c71d8a6)

mj22226 committed:
mac80211: realtek: rtw88: sync with v6.16

Renamed 046..051 pending patches with the merged tag/hash.
Two patches were skipped, they make use of WQ_BH present since v6.14:
  https://github.com/torvalds/linux/commit/13221be72034d1c34630ab124c43438aefe7e656
  https://github.com/torvalds/linux/commit/3e3aa566dd1803f1697530de6c8489a8350765b3
Manually refreshed 062-v6.14-wifi-rtw88-Add-support-for-LED-blinking.patch
Manually refreshed  063-v6.14-wifi-rtw88-add-RTW88_LEDS-depends-on-LEDS_CLASS-to-K.patch
Manually backported 090-v6.15-wifi-rtw88-Enable-the-new-RTL8814AE-RTL8814AU-driver.patch

git log --no-merges --pretty=oneline --abbrev-commit 4c2c372de...0daa521a drivers/net/wireless/realtek/rtw88:
4c2c372de2e1 wifi: rtw88: fix the 'para' buffer size to avoid reading out of bounds
f24d0d8c3cd7 wifi: rtw88: Fix the random "error beacon valid" messages for USB
80fe0bc1659c wifi: rtw88: usb: Upload the firmware in bigger chunks
490340faddea wifi: rtw88: usb: Reduce control message timeout to 500 ms
b7f0cc647e52 wifi: rtw88: rtw8822bu VID/PID for BUFFALO WI-U2-866DM
2c17afde9ff6 wifi: rtw88: Handle RTL8723D(S) with blank efuse
0ffa1ba81b35 wifi: rtw88: Fix RX aggregation settings for RTL8723DS
20d3c19bd8f9 wifi: rtw88: do not ignore hardware read error during DPK
fc5f5a0ec463 wifi: rtw88: sdio: call rtw_sdio_indicate_tx_status unconditionally
b2effcdc2379 wifi: rtw88: sdio: map mgmt frames to queue TX_DESC_QSEL_MGMT
581cf3a9cb61 wifi: rtw88: Fix the module names printed in dmesg
b8d49bb8d16a wifi: rtw88: Don't set SUPPORTS_AMSDU_IN_AMPDU for RTL8814AU
0d2a88690e58 wifi: rtw88: Set AMPDU factor to hardware for RTL8814A
dcbb7bb3a364 wifi: rtw88: usb: Enable RX aggregation for RTL8814AU
bf1103654df9 wifi: rtw88: usb: Enable switching the RTL8814AU to USB 3
625fbc16524a wifi: rtw88: usb: Remove redundant 'flush_workqueue()' calls
5c4cf36c538b wifi: rtw88: sdio: Remove redundant 'flush_workqueue()' calls
d58ad77d5cc2 wifi: rtw88: Add __nonstring annotations for unterminated strings
deb3ddeb1865 wifi: rtw88: Enable the new RTL8814AE/RTL8814AU drivers
bad060e8a425 wifi: rtw88: Add rtw8814au.c
dad8e8793102 wifi: rtw88: Add rtw8814ae.c
1a7545784642 wifi: rtw88: Add rtw8814a.{c,h}
e38246889cc9 wifi: rtw88: Add rtw8814a_table.c (part 2/2)
f4debfcb1b3c wifi: rtw88: Add rtw8814a_table.c (part 1/2)
679ec431477c wifi: rtw88: Add some definitions for RTL8814AU
c374281f8285 wifi: rtw88: Extend rtw_debugfs_get_tx_pwr_tbl() for RTL8814AU
cfebabdd351e wifi: rtw88: Extend rtw_debugfs_get_phy_info() for RTL8814AU
8b42c46cf665 wifi: rtw88: Extend rtw_phy_config_swing_table() for RTL8814AU
053a7aace020 wifi: rtw88: Fix rtw_rx_phy_stat() for RTL8814AU
6be7544d19fc wifi: rtw88: Fix rtw_init_vht_cap() for RTL8814AU
c7eea1ba05ca wifi: rtw88: Fix rtw_init_ht_cap() for RTL8814AU
86d04f8f991a wifi: rtw88: Fix rtw_desc_to_mcsrate() to handle MCS16-31
e66bca16638e wifi: rtw88: Fix rtw_mac_power_switch() for RTL8814AU
80c4668d024f wifi: rtw88: Add support for Mercusys MA30N and D-Link DWA-T185 rev. A1
9f00e2218e15 wifi: rtw88: Fix rtw_update_sta_info() for RTL8814AU
0f98a5959657 wifi: rtw88: Extend TX power stuff for 3-4 spatial streams
ad815f392003 wifi: rtw88: Rename RTW_RATE_SECTION_MAX to RTW_RATE_SECTION_NUM
e66f3b5c7535 wifi: rtw88: Constify some more structs and arrays
8f0076726b66 wifi: rtw88: Extend rtw_fw_send_ra_info() for RTL8814AU
d80e7d9b6ba3 wifi: rtw88: Extend rf_base_addr and rf_sipi_addr for RTL8814AU
62f726848da4 wifi: rtw88: Extend struct rtw_pwr_track_tbl for RTL8814AU
9e8243025cc0 wifi: rtw88: Fix download_firmware_validate() for RTL8814AU
8425f5c8f04d wifi: rtw88: Fix __rtw_download_firmware() for RTL8814AU
105dc94233e4 wifi: rtw88: Fix a typo of debug message in rtw8723d_iqk_check_tx_failed()
0d1d165eff9d wifi: rtw88: Don't use static local variable in rtw8821c_set_tx_power_index_by_rate
00451eb3bec7 wifi: rtw88: Don't use static local variable in rtw8822b_set_tx_power_index_by_rate
b4bfbc50b1b9 wifi: rtw88: add RTW88_LEDS depends on LEDS_CLASS to Kconfig
4b6652bc6d8d wifi: rtw88: Add support for LED blinking
fb2fcfbe5eef wifi: rtw88: sdio: Fix disconnection after beacon loss
a806a8160a0f wifi: rtw88: 8703b: Fix RX/TX issues
5ad483955acc wifi: rtw88: Delete rf_type member of struct rtw_sta_info
5b1b9545262b wifi: rtw88: Add USB PHY configuration
not backported (3e3aa566dd18 wifi: rtw88: usb: Preallocate and reuse the RX skbs)
not backported (13221be72034 wifi: rtw88: Handle C2H_ADAPTIVITY in rtw_fw_c2h_cmd_handle() )
e9048e2935f7 wifi: rtw88: usb: Copy instead of cloning the RX skb
74a72c367573 wifi: rtw88: 8821a/8812a: Set ptct_efuse_size to 0
59ab27a9f20f wifi: rtw88: 8812a: Support RFE type 2

Signed-off-by: Marty Jones <mj8263788@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/19052
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>
(cherry picked from commit 4a1aca53761643404a81f8023eb16b53142423df)

georgemoussalem committed:
wifi: ath11k: disable coldboot calibration for ipq5018 and qcn6122

Coldboot calibration does not work causes the firmware to crash during
wifi startup. So let's disable coldboot calibration until a solution is
found.

Signed-off-by: George Moussalem <george.moussalem@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/19083
Signed-off-by: Robert Marko <robimarko@gmail.com>
(cherry picked from commit cf715a230589058aeb703f930504d6d98619a53b)

georgemoussalem committed:
wifi: ath11k: increase max ATH11K_QMI_CALDB_SIZE macro

QCN6122 wifi in fw-memory-mode 1 has a slightly larger caldb size than
currently defined in the ath11k driver. When coldboot calibration was
disabled, the fw mem mode was changed from 2 (256MB mem profile) to 1
(512MB mem profile), which is the correct setting for devices in scope.
However, in fw mem mode, the caldb size is 0x500000 instead of the max
0x480000 defined in the driver, causing QCN6122 wifi failing to boot:

ath11k b00a040.wifi1: qmi mem size is low to load caldata
ath11k b00a040.wifi1: failed to assign qmi target memory: -22

As such, change the max caldb memory size accordingly.

This macro is used by the driver only as a max size limit to validate
the requested caldb size returned by QMI. Different ath11k wifi chips
have different caldb sizes (for ex. the size for IPQ5018 is 0x200000).

Fixes: cf715a230589 ("wifi: ath11k: disable coldboot calibration for ipq5018")
Signed-off-by: George Moussalem <george.moussalem@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/19118
Signed-off-by: Robert Marko <robimarko@gmail.com>
(cherry picked from commit aefdbf0edc9ea046f55868ddee85442a591b2e4b)

981213 commented:
We don't do mac80211 major version bumps for stable releases unfortunately. This package is treated the same as the kernel itself as it replaces the wifi stack and all wifi drivers.

nbd168 committed:
mac80211: fix rt2x00 build error on 6.12

Fix missing remove callback type signature conversion

Signed-off-by: Felix Fietkau <nbd@nbd.name>
(cherry picked from commit e1d39bdbdb0fabbe6adb32f1ce69797ce4248dc4)

nbd168 committed:
ath9k: add back previous DT compatible strings for ahb

The ath9k ahb patch was updated to match the latest upstream version,
however the openwrt DT files still use the older names.
Add those as extra entries in order to remain compatible until DT files
are fixed.

Fixes: 88f4c32060f7 ("mac80211: update to version 6.14.11")
Signed-off-by: Felix Fietkau <nbd@nbd.name>
(cherry picked from commit d2b2670937f051bda7c8fbda8a6c0131977d5e68)

andreybondar59 commented:
Would you please add monitor interface support on QCN9274 ?

namiltd commented:
Continued at https://github.com/namiltd/openwrt/pull/87
